alternator/executor.cc

```diff
@@ -2391,7 +2396,7 @@ rmw_operation::write_isolation rmw_operation::get_write_isolation_for_schema(sch
 // a read-before-write, but not just on it - depending on configuration,
 // execute() may unconditionally use cas() for every write. Unfortunately,
 // this requires duplicating here a bit of logic from execute().
-std::optional<shard_id> rmw_operation::shard_for_execute(bool needs_read_before_write) {
+std::optional<shard_id> rmw_operation::shard_for_execute(locator::erm_handle, bool needs_read_before_write) {
```

Comment:
It seems like you're not using this new erm_handle argument in this function, so why add it to the function?

Reply:
May be `shard_for_execute` ca return `erm_handle` instead?

Reply:
The message it conveys is that the decision was made using the topology `erm_handle` now holds and the decision is only valid while it is held. It does not matter that sometimes the decision is "I do not care". We can also  have `erm_handle` handler type that can hold nothing, but if we want to use this for regular requests in the future as well why not just return valid one from the start.

Reply:
OK